xmtp/src/codecs/text.rs

```diff
@@ -0,0 +1,66 @@
+use super::{CodecError, ContentCodec};
+use std::collections::HashMap;
+use xmtp_proto::xmtp::message_contents::{ContentTypeId, EncodedContent};
+
+pub struct TextCodec {}
+impl TextCodec {
+    const AUTHORITY_ID: &str = "xmtp.org";
+    const TYPE_ID: &str = "text";
+    const ENCODING_KEY: &str = "encoding";
+    const ENCODING_UTF8: &str = "UTF-8";
+}
+
+impl ContentCodec<String> for TextCodec {
+    fn content_type() -> ContentTypeId {
+        ContentTypeId {
+            authority_id: TextCodec::AUTHORITY_ID.to_string(),
+            type_id: TextCodec::TYPE_ID.to_string(),
+            version_major: 1,
+            version_minor: 0,
+        }
+    }
+
+    fn encode(text: String) -> Result<EncodedContent, CodecError> {
+        Ok(EncodedContent {
+            r#type: Some(TextCodec::content_type()),
```

Comment:
Am I correct that a raw string literal is used here because type is a keyword?

Reply:
Found my own answer. This is not a raw string literal, as the grammar does not match. This is a "keyword Identifier"